src/libraries/System.Security.Cryptography.Cose/tests/CoseTestHelpers.cs

```diff
@@ -247,6 +289,35 @@ private static void AssertSign1Headers(CborReader reader, List<(CoseHeaderLabel,
         internal static RSA RSAKey => t_rsaKey ??= CreateRSA(true);
         internal static RSA RSAKeyWithoutPrivateKey => t_rsaKeyWithoutPrivateKey ??= CreateRSA(false);
 
+        internal static List<(RSA, HashAlgorithmName)> RSAKeyHashPairs = new()
+        {
+            (RSAKey, HashAlgorithmName.SHA256),
+            (RSAKey, HashAlgorithmName.SHA384),
+            (RSAKey, HashAlgorithmName.SHA512),
+        };
+
+        internal static List<(RSA, HashAlgorithmName)> RSANonPrivateKeyHashPairs = new()
+        {
+            (RSAKeyWithoutPrivateKey, HashAlgorithmName.SHA256),
+        };
+
+        internal static List<(AsymmetricAlgorithm, HashAlgorithmName)> ConcatECDsaAndRSAKeyHashPairs(
```

Comment:
This is where you broke your `[ThreadStatic]`.  You promoted the keys into a List, which moves them into the same key object for all tests.